### PR TITLE
feat: android sdk connectsign + batch request

### PR DESCRIPTION
**Pull request by abretonc7s**
## **Description**

- Depends on Connection Refactor PR https://github.com/MetaMask/metamask-mobile/pull/7895
- Adds android sdk `metamask_connectSign` + `metamask_batch`
- Improve loading status

## **Related issues**

Fixes: #

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/core/SDKConnect/SDKConnect.ts [hunk 1]
```diff
@@ -270,32 +261,30 @@ export class SDKConnect extends EventEmitter2 {
     loading: boolean;
   }) {
     if (loading === true) {
-      DevLogger.log(``);
+      this.sdkLoadingState[channelId] = true;
+    } else {
+      delete this.sdkLoadingState[channelId];
+    }
+
+    const loadingSessions = Object.keys(this.sdkLoadingState).length;
```
Comment (omridan159) [inline app/core/SDKConnect/SDKConnect.ts:269]:
```suggestion
    const  amountOfLoadingSessions = Object.keys(this.sdkLoadingState).length;
```

File: app/core/SDKConnect/handlers/checkPermissions.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import { Json } from '@metamask/utils';
+
+// TODO: should be more generic and be used in wallet connect and android service as well
+export const checkPermissions = async ({
+  connection,
+  Engine,
+  lastAuthorized,
+}: {
+  connection: Connection;
+  Engine: any;
```
Comment (omridan159) [inline app/core/SDKConnect/handlers/checkPermissions.ts:18]: Change to the real type

File: app/core/SDKConnect/handlers/handleConnectionMessage.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+import handleCustomRpcCalls from './handleCustomRpcCalls';
+import handleSendMessage from './handleSendMessage';
+
+export const handleConnectionMessage = async ({
+  message,
+  Engine,
+  connection,
+}: {
+  message: CommunicationLayerMessage;
+  Engine: any;
```
Comment (omridan159) [inline app/core/SDKConnect/handlers/handleConnectionMessage.ts:24]: Change to the real type

File: app/core/SDKConnect/handlers/handleConnectionReady.ts [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+export const handleConnectionReady = async ({
+  originatorInfo,
+  Engine,
+  connection,
+  approveHost,
+  disapprove,
+  updateOriginatorInfos,
+}: {
+  originatorInfo: OriginatorInfo;
+  Engine: any;
```
Comment (omridan159) [inline app/core/SDKConnect/handlers/handleConnectionReady.ts:24]: Change to the real type

**Review by omridan159 — DISMISSED** [review]

**Review by andreahaku — APPROVED** [review]
LGTM

**Pull request merged**